Document/0x07c-Testing-Cryptography.md

```diff
@@ -2,14 +2,33 @@
 
 The following chapter translates the cryptography requirements of the MASVS into technical test cases. Test cases listed in this chapter are based upon generic cryptographic concepts and are not relying on a specific implementation on iOS or Android.
 
-Proper design of a cryptographic system is a common pitfall for mobile application development. To achieve good security, a developer has to chose the right cryptographic directive (e.g., symmetric encryption), chose the right implementation for that directive (e.g., AES-GCM) and then configure that implementation correctly (e.g., key length, block modes, key management). While this chapter does not give an introduction into cryptography, its questions are designed to find common problems within the mentioned selection and implementation process.
+The primary goal of cryptography is to provide confidentiality, data integrity, and authentication, even in the presence of a malicious attacker. Confidentiality is achieved through use of encryption, with the aim of ensuring secrecy of the contents. Data integrity prevents tampering or modifying the data, and authentication ensures that the data came from a trusted source.
```

Comment:
> Data integrity prevents tampering or modifying the data

I don't think this is entirely correct. "Data integrity" is more of a concept, and not really a practice. As such, it does not prevent anything.